src/Type/BaseDoctrineORMSerializationType.php

```diff
@@ -83,15 +88,17 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
 
         $manager = $this->registry->getManagerForClass($this->class);
         $doctrineMetadata = $manager->getClassMetadata($this->class);
+        \assert($doctrineMetadata instanceof ClassMetadataInfo);
 
         foreach ($serializerMetadata->propertyMetadata as $propertyMetadata) {
-            $name = $propertyMetadata->name;
+            \assert($propertyMetadata instanceof PropertyMetadata);
 
+            $name = $propertyMetadata->name;
             if (\in_array($name, $doctrineMetadata->getIdentifierFieldNames(), true) && !$this->identifierOverwrite) {
                 continue;
             }
 
-            if (!$propertyMetadata->groups || !\in_array($this->group, $propertyMetadata->groups, true)) {
```

Comment:
After this change all entity fields must have groups defined . If not, the "in_array() expects parameter 2 to be array, null given" exception is thrown.

Reply:
I followed the phpdoc of https://github.com/schmittjoh/serializer/blob/master/src/Metadata/PropertyMetadata.php#L25

Do you mean the phpdoc is wrong @mareckigit ?

Then, the best would be to 
- either update the phpdoc to `string[]|null` on this package in order to avoid phpstan error
- either add a default value to `[]`

Reply:
It looks like the doc is wrong. public $groups is not initialized, so it's null. For example not all fields in Sonata Media Entity have set groups (https://github.com/sonata-project/SonataMediaBundle/blob/3.x/src/Resources/config/serializer/Model.Media.xml).

Reply:
But not initialized is not the same as null, it can be string the property but not be initialized. Or am I missing something?

Reply:
Well, if the property is not typehinted as string, it is the same. (it wouldn't be otherwise). So the phpdoc is indeed wrong.

Reply:
I created https://github.com/schmittjoh/serializer/issues/1371

Reply:
And this should fix the bug https://github.com/sonata-project/form-extensions/pull/296